api/src/main/java/io/grpc/HasByteBuffer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+
+/**
+ * Extension to an {@link java.io.InputStream} whose content can be accessed as {@link
+ * ByteBuffer}s.
+ *
+ * <p>This can be used for optimizing the case for the consumer of a {@link ByteBuffer}-backed
+ * input stream supports efficient reading from {@link ByteBuffer}s directly. This turns the reader
+ * interface from an {@link java.io.InputStream} to {@link ByteBuffer}s, without copying the
+ * content to a byte array and read from it.
+ */
+// TODO(chengyuanzhang): add ExperimentalApi annotation.
+public interface HasByteBuffer {
+
+  /**
+   * Indicates whether or not {@link #getByteBuffer} operation is supported.
+   */
+  boolean getByteBufferSupported();
```

Comment:
Why not give this the same name as the equivalent added `ReadableBuffer` method (my suggestion would be `hasByteBuffer()`)?

Reply:
It doesn't need to match what we use for `ReadableBuffer` right? We'd probably have a discussion on this API this Thursday and we usually do a vote for the name. 

`hasByteBuffer` 1 vote now.

Reply:
As well as for consistency, having it line up with `ReadableBuffer` would permit `ReadableBuffer`s to themselves be `InputStream`s implementing this interface, which I think might allow for some other streamlining later.

Reply:
Maybe, for that reason.

My original thought was this `HasByteBuffer` interface should only have the `getByteBuffer` method. InputStreams not supporting the operation should just not implement this interface. https://github.com/grpc/grpc-java/pull/7330#discussion_r471594993 suggests combining two InputStream implementations into one by adding this `getByteBufferSupported` method. This look ok, but making the method `hasByteBuffer` does make this API wired. I'd rather change the `hasByteBuffer` method on `ReadableBuffer` interface to `getByteBufferSupported`.